src/common/BaseCommand.ts

```diff
@@ -52,23 +52,17 @@ export abstract class BaseCommand
             .then(() => this.parse())
             .then((parserOutput) => {
                 this._flagValues = parserOutput.flags;
-                return Logger.child(this.commandName);
+                return new Logger(this.commandName);
             })
             .then((logger) => {
                 // extract the log level flag (if any) and
                 // set the logger's level to this value
-                const logLevelFlag: string | undefined = (
-                    this._flagValues.loglevel as string
-                )
-                    ?.trim()
-                    ?.toUpperCase();
-                const logLevel =
```

Comment:
This was doing the wrong thing. Casting as `(<any>LoggerLevel)[logLevelFlag]` was returning a `string` value instead of a `LoggerLevel` value so then the call to `logger.setLevel(logLevel);` which expects a `LoggerLevel` value would not accept the input param and instead would choose a default log level no matter what the actual value of `logLevel` was. Fixing this here using a helper function to get a `LogLevel` representation of the value.